internal/otelmatchers/matchers.go

```diff
@@ -0,0 +1,84 @@
+package otelmatchers
+
+import (
+	"fmt"
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func ConsistOfSpansWithIDs(spanIDs []string) types.GomegaMatcher {
```

Comment:
I think the names could be more descriptive. The name`spanIDs` is used 2 times (in the `WithTransform` and here) with different meanings. Here it describes the expected Span IDs and in the `WithTransform` it describes the actual Span IDs.  So something like `expectedSpanIDs` and `actualSpanIDs` would help understanding the code :)

Reply:
Agree, done